ev3dev2/motor.py

```diff
@@ -1798,6 +1828,9 @@ def __init__(self, left_motor_port, right_motor_port, desc=None, motor_class=Lar
         self.right_motor = self.motors[right_motor_port]
         self.max_speed = self.left_motor.max_speed
 
+        # color sensor used by follow_line()
+        self.cs = None
```

Comment:
What's the thought process behind building this into `MoveTank` rather than having a separate class? I feel like having a `LineFollower` class which can be constructed with a custom `MoveTank` instance if needed would be better, even if just for code separation. I'm a bit scared by classes like this gaining a many "optional" features which have to coexist.

Reply:
yeah something I have been contemplating is do we just make a `FirstLegoLeagueRobot` class/file (open for suggestions on the name) where we have a child class of `MoveTank` that holds these sorts of things.  All of the odometry functionality in `MoveTank` could go here as well.

Reply:
Why not just make individual classes for each behavior, so I can do this?

```python
drive = MoveTank(OUTPUT_A, OUTPUT_B)

# get other utilities if I want them
line_follower = PidLineFollower(drive)
odometry = Odometry(drive) # or just take in motors directly

line_follower.follow_line()
# use odometry...

```

Reply:
I just tend to think in terms of adding functionality via methods in child classes.  To me that is cleaner and easier to follow than having a standalone function that can only accept certain types of objects to implement the desired functionality.